units/commit_queue.go

```diff
@@ -161,6 +189,22 @@ func (j *commitQueueJob) Run(ctx context.Context) {
 	})
 }
 
+func IsPatchDone(patch_id string) (bool, error) {
+	patch, err := patch.FindOne(patch.ById(patch.NewId(patch_id)))
+	if err != nil {
+		return false, err
+	}
+	finishTime := patch.FinishTime
```

Comment:
you don't need to assign this to another variable, and assigning to an isDone is maybe more extrapolation than is worth: you could simplify all of 197 to 205 with `return !patch.FinishTime.IsZero(), nil`  (or at least return in the if statement and drop the else).

Also I definitely commend modular functions 🎉  but this function and the one calling it are both short enough that I think combining them would be fine.